lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -85,55 +134,14 @@ private void Initialize()
         /// <returns></returns>
         public T GetPartitionHandle()
         {
-            if (_copyToLocal)
-            {
-                if (!_isInitialized)
-                {
-                    Initialize();
-                }
-
-                return _fileSerializer.Deserialize(_localFiles);
-            }
-            return _fileSerializer.Deserialize(_filePaths);
-        }
-
-        private void CopyFromRemote()
-        {
-            string localFileFolder = _tempFileCreator.CreateTempDirectory("-partition-");
-            Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "Local file temp folder: {0}", localFileFolder));
-
-            foreach (var sourceFilePath in _filePaths)
+            lock (_lock)
             {
-                Uri sourceUri = _fileSystem.CreateUriForPath(sourceFilePath);
-                Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "sourceUri {0}: ", sourceUri));
-                if (!_fileSystem.Exists(sourceUri))
+                if (!_localFiles.IsPresent())
```

Comment:
This just makes sure that if user has not called AddCache() already call it now and also make sure that logic remains same for old users? i..e they do not need AddCache()?